src/qat/ir/instructions.py

```diff
@@ -0,0 +1,1032 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from pydoc import locate
+from typing import Any, Dict, List, Literal, Optional, Set, Union
+
+import numpy as np
+from compiler_config.config import InlineResultsProcessing
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    ValidationInfo,
+    field_serializer,
+    field_validator,
+)
+from typing_extensions import Annotated
+
+from qat.purr.compiler.devices import PulseChannel, PulseShapeType, QuantumComponent, Qubit
+
+# The following things from legacy instructions are unchanged, so just import for now.
+from qat.purr.compiler.instructions import AcquireMode, PostProcessType, ProcessAxis
+from qat.purr.utils.logger import get_default_logger
+
+
+### Sweeps
+class SweepOperation(BaseModel):
+    """Common parent for all things that need differentiating during a sweep."""
+
+    pass
+
+
+class SweepValue(SweepOperation):
+    name: str
+    value: Any
+
+
+### Variables
+def build_generated_name(existing_names=None, prefix=None):
```

Comment:
Why do we need to generate random names, can't we just use uuids?

Reply:
Yeah would probably be easier (and safer) to use UUIDs. Not sure if there is anywhere else which is expected it to have a name with this structure though... Nevertheless, I haven't changed this from the legacy so I'll just change this to import for now